astropy/stats/lombscargle/implementations/cython_impl.pyx

```diff
@@ -0,0 +1,252 @@
+        case of incomplete phase coverage.
+    center_data : bool (optional, default=True)
+        if True, pre-center the data by subtracting the weighted mean
+        of the input data. This is especially important if ``fit_bias = False``
+
+    Returns
+    -------
+    power : array_like
+        Lomb-Scargle power associated with each frequency.
+        Units of the result depend on the normalization.
+
+    References
+    ----------
+    .. [1] W. Press et al, Numerical Recipies in C (2002)
+    .. [2] Scargle, J.D. 1982, ApJ 263:835-853
+    .. [3] M. Zechmeister and M. Kurster, A&A 496, 577-584 (2009)
+    """
+    if dy is None:
+        dy = 1
+
+    t, y, dy = np.broadcast_arrays(t, y, dy)
+    t = np.asarray(t, dtype=DTYPE, order='C')
+    y = np.asarray(y, dtype=DTYPE, order='C')
+    dy = np.asarray(dy, dtype=DTYPE, order='C')
+    frequency = np.asarray(frequency, dtype=DTYPE, order='C')
+
+    if t.ndim != 1:
+        raise ValueError("t, y, dy should be one dimensional")
+    if frequency.ndim != 1:
+        raise ValueError("frequency should be one-dimensional")
+
+    PLS = np.zeros(frequency.shape, dtype=DTYPE, order='C')
+
+    # pre-center the data: not technically required if fit_bias=True,
+    # but it simplifies the math.
+    if fit_bias or center_data:
+        # compute MLE for mean in the presence of noise.
+        w = dy ** -2
+        y = y - np.dot(w, y) / np.sum(w)
+
+    if fit_bias:
+        _generalized_lomb_scargle(t, y, dy, 2 * np.pi * frequency, PLS)
+    else:
+        _standard_lomb_scargle(t, y, dy, 2 * np.pi * frequency, PLS)
+
+    if normalization == 'standard':
+        pass
+    elif normalization == 'model':
+        return PLS / (1 - PLS)
+    elif normalization == 'log':
+        return -np.log(1 - PLS)
+    elif normalization == 'psd':
+        w = dy ** -2
+        w /= w.sum()
+        PLS *= 0.5 * t.size * np.dot(w, y * y)
+    else:
+        raise ValueError("normalization='{0}' "
+                         "not recognized".format(normalization))
+    return PLS.ravel()
+
+
+@cython.cdivision(True)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef _standard_lomb_scargle(DTYPE_t[::1] t, DTYPE_t[::1] y, DTYPE_t[::1] dy,
+                            DTYPE_t[::1] omega, DTYPE_t[::1] PLS):
+    cdef ITYPE_t N_freq = omega.shape[0]
+    cdef ITYPE_t N_obs = t.shape[0]
+
+    cdef DTYPE_t w, omega_t, sin_omega_t, cos_omega_t, tan_2omega_tau
+    cdef DTYPE_t S2, C2, tau, Y, wsum, YY, YCtau, YStau, CCtau, SStau
+
+    for i in range(N_freq):
+        # first pass: determine tau
+        S2 = 0
+        C2 = 0
+        for j in range(N_obs):
+            w = 1. / dy[j]
+            w *= w
+
+            omega_t = omega[i] * t[j]
+            sin_omega_t = sin(omega_t)
+            cos_omega_t = cos(omega_t)
+
+            S2 += 2 * w * sin_omega_t * cos_omega_t
+            C2 += w * (1 - 2 * sin_omega_t * sin_omega_t)
+
+        tau = 0.5 * atan(S2 / C2) / omega[i]
```

Comment:
I think in this case the quadrant won't matter, because tau is only used directly as a phase within a sine or cosine that is eventually squared.